pkg/controller/cloudfront/cloudfrontoriginaccessidentity/setup.go

```diff
@@ -0,0 +1,120 @@
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		WithOptions(controller.Options{
+			RateLimiter: ratelimiter.NewController(rl),
+		}).
+		For(&svcapitypes.CloudFrontOriginAccessIdentity{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(svcapitypes.CloudFrontOriginAccessIdentityGroupVersionKind),
+			managed.WithExternalConnecter(&connector{
+				kube: mgr.GetClient(),
+				opts: []option{
+					func(e *external) {
+						e.preObserve = preObserve
+						e.postObserve = postObserve
+						e.preCreate = preCreate
+						e.postCreate = postCreate
+						e.lateInitialize = lateInitialize
+						e.preUpdate = preUpdate
+						e.isUpToDate = isUpToDate
+						e.preDelete = preDelete
+					},
+				},
+			}),
+			managed.WithPollInterval(poll),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+func preCreate(_ context.Context, cr *svcapitypes.CloudFrontOriginAccessIdentity, cdi *svcsdk.CreateCloudFrontOriginAccessIdentityInput) error {
+	cdi.CloudFrontOriginAccessIdentityConfig.CallerReference = awsclients.String(string(cr.UID))
+	return nil
+}
+
+func postCreate(_ context.Context, cp *svcapitypes.CloudFrontOriginAccessIdentity, cpo *svcsdk.CreateCloudFrontOriginAccessIdentityOutput,
+	ec managed.ExternalCreation, err error) (managed.ExternalCreation, error) {
+	if err != nil {
+		return managed.ExternalCreation{}, err
+	}
+
+	meta.SetExternalName(cp, awsclients.StringValue(cpo.CloudFrontOriginAccessIdentity.Id))
+	return ec, nil
+}
+
+func preObserve(_ context.Context, cp *svcapitypes.CloudFrontOriginAccessIdentity, gpi *svcsdk.GetCloudFrontOriginAccessIdentityInput) error {
+	gpi.Id = awsclients.String(meta.GetExternalName(cp))
+	return nil
+}
+
+func postObserve(_ context.Context, cp *svcapitypes.CloudFrontOriginAccessIdentity, _ *svcsdk.GetCloudFrontOriginAccessIdentityOutput,
+	eo managed.ExternalObservation, err error) (managed.ExternalObservation, error) {
+	if err != nil {
+		return managed.ExternalObservation{}, err
+	}
+	cp.SetConditions(xpv1.Available())
+	return eo, nil
+}
+
+func preUpdate(_ context.Context, cp *svcapitypes.CloudFrontOriginAccessIdentity, upi *svcsdk.UpdateCloudFrontOriginAccessIdentityInput) error {
+	upi.CloudFrontOriginAccessIdentityConfig.CallerReference = awsclients.String(string(cp.UID))
+	upi.Id = awsclients.String(meta.GetExternalName(cp))
+	upi.SetIfMatch(awsclients.StringValue(cp.Status.AtProvider.ETag))
+	return nil
+}
+
+func preDelete(_ context.Context, cp *svcapitypes.CloudFrontOriginAccessIdentity, dpi *svcsdk.DeleteCloudFrontOriginAccessIdentityInput) (bool, error) {
+	dpi.Id = awsclients.String(meta.GetExternalName(cp))
+	dpi.SetIfMatch(awsclients.StringValue(cp.Status.AtProvider.ETag))
+	return false, nil
+}
+
+func lateInitialize(in *svcapitypes.CloudFrontOriginAccessIdentityParameters, gpo *svcsdk.GetCloudFrontOriginAccessIdentityOutput) error {
```

Comment:
@stevendborrelli  small nit pick: https://github.com/crossplane/provider-aws/blob/e6d9e8073e098c714edccdea331f5a77e262dc9d/pkg/controller/cloudfront/cloudfrontoriginaccessidentity/zz_controller.go#L226 
can you remote lateInitialize its a noOps currently in generated code =)

Reply:
removed in 603cb5b5.